internal/service/advancedclustertpf/resource_upgrade.go

```diff
@@ -0,0 +1,27 @@
+package advancedclustertpf
+
+import (
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/constant"
+	"go.mongodb.org/atlas-sdk/v20241113001/admin"
+)
+
+func tenantUpgrade(state, patch *admin.ClusterDescription20240805) *admin.LegacyAtlasTenantClusterUpgradeRequest {
+	if patch.ReplicationSpecs == nil {
+		return nil
+	}
+	oldRegion := state.GetReplicationSpecs()[0].GetRegionConfigs()[0]
+	oldProviderName := oldRegion.GetProviderName()
+	newRegion := patch.GetReplicationSpecs()[0].GetRegionConfigs()[0]
+	newProviderName := newRegion.GetProviderName()
+	if oldProviderName != constant.TENANT || newProviderName == constant.TENANT {
```

Comment:
A bit simplified logic than [existing](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/internal/service/advancedcluster/resource_advanced_cluster.go#L1331), but better IMO:
Not depending on knowing shared instance sizes.
Should always be an upgrade when changing from TENANT -> AWS|GCP|AZURE

Reply:
Are we sure this change in logic is equivalent to the existing one?

Reply:
I'm open to change this based on evidence :D